### PR TITLE
Bump parent pom to 3.54 [v2]

### PR DESCRIPTION
**Pull request by slonopotamus**
This PR is a v2 of #90. It bumps subversion-plugin to 2.13.0 instead of adding a workaround for jenkinsci/subversion-plugin#240

### CODE REVIEW

**Comment by slonopotamus** [comment]
@jglick Hmm... `SCMBinderTest.exactRevisionSubversion` [doesn't pass on Linux](https://ci.jenkins.io/job/Plugins/job/workflow-multibranch-plugin/job/PR-91/2/testReport/junit/org.jenkinsci.plugins.workflow.multibranch/SCMBinderTest/exactRevisionSubversion/).

File: pom.xml [hunk 1]
```diff
@@ -66,8 +66,7 @@ THE SOFTWARE.
         <revision>2.22</revision>
         <changelist>-SNAPSHOT</changelist>
         <java.level>8</java.level>
-        <jenkins.version>2.121.1</jenkins.version>
```
Comment (jglick) [inline pom.xml]: You should always specify an explicit `jenkins.version`. Do _not_ attempt to inherit from the parent POM; that is merely a minimum recommended version.

**Comment by jglick** [comment]
Seems that `SCMBinderTest.exactRevisionSubversion` passes with a `subversion` dep up to `2.7.2`. From `2.8`–`2.11.1` it fails with

```
org.junit.ComparisonFailure: expected:<[[/prj/trunk/Jenkinsfile, /prj/trunk/]file]> but was:<[[Jenkinsfile, ]file]>
	at org.junit.Assert.assertEquals(Assert.java:115)
	at org.junit.Assert.assertEquals(Assert.java:144)
	at org.jenkinsci.plugins.workflow.multibranch.SCMBinderTest.exactRevisionSubversion(SCMBinderTest.java:167)
```

and from `2.12.0`–`2.13.0` it fails with

```
java.lang.IndexOutOfBoundsException: Index: 0, Size: 0
	at java.util.ArrayList.rangeCheck(ArrayList.java:657)
	at java.util.ArrayList.get(ArrayList.java:433)
	at org.jenkinsci.plugins.workflow.multibranch.SCMBinderTest.exactRevisionSubversion(SCMBinderTest.java:160)
```

I will dig into this a bit and probably propose a move of the SVN-based tests out of here. (They do exercise some aspects of Pipeline/SCM integration that the Git plugin does not, due to the very different model of versioning, but probably not enough to be worth the hassle of testing against what is increasingly an obsolete tool.)

File: pom.xml (hunk 1 above)
Comment (slonopotamus) [reply pom.xml]: Ok, reverted this change.

File: pom.xml [hunk 2]
```diff
@@ -67,7 +67,7 @@ THE SOFTWARE.
         <changelist>-SNAPSHOT</changelist>
         <java.level>8</java.level>
         <jenkins.version>2.121.1</jenkins.version>
```
Comment (jglick) [inline pom.xml]: May work but 2.138.4 is the minimum recommended version.

File: pom.xml [hunk 3]
```diff
@@ -201,7 +201,7 @@ THE SOFTWARE.
         <dependency>
             <groupId>org.jenkins-ci.plugins</groupId>
             <artifactId>subversion</artifactId>
-            <version>2.7.1</version>
+            <version>2.13.0</version>
```
Comment (jglick) [inline pom.xml]: Need to introduce a POM property so that the `tests` JAR version (line ~226) is also updated.

Comment (slonopotamus) [reply pom.xml]: Good catch, I missed that.

File: pom.xml (hunk 2 above)
Comment (slonopotamus) [reply pom.xml]: Okay.

File: pom.xml (hunk 3 above)
Comment (slonopotamus) [reply pom.xml]: Done

**Comment by jglick** [comment]
BTW please do not force-push inside a PR branch! It breaks the GH review flow, makes regular `git` CLI operations like `merge` and `pull` break in crazy ways, etc. The repository maintainer may choose to squash-merge or whatever if and when the PR is accepted, but leave that decision to them.

**Comment by jglick** [comment]
also need

```diff
         <dependency>
             <groupId>org.tmatesoft.svnkit</groupId>
             <artifactId>svnkit-cli</artifactId>
-            <version>1.8.14</version>
+            <version>1.10.1</version>
             <scope>test</scope>
         </dependency>
         <dependency>
```

**Comment by jglick** [comment]
The earlier test failure is probably from https://github.com/jenkinsci/subversion-plugin/pull/183 (unconfirmed), which seems harmless enough, but the later one looks like a real regression so far. I will continue working to track it down as de facto maintainer (not that I want to be) but I think this PR is hopeless to continue as such; I will likely just wind up filing a fresh PR to handle the test split once the root causes are understood.

**Comment by jglick** [comment]
OK that might be an overreaction. The second failure looks to be from https://github.com/jenkinsci/subversion-plugin/pull/223 and is unavoidable pending more complex changes to `scm-api` to allow lightweight changelog calculation. So for now it suffices, in addition to the `svnkit-cli` bump above (needed to match upstream), to do

```diff
diff --git src/test/java/org/jenkinsci/plugins/workflow/multibranch/SCMBinderTest.java src/test/java/org/jenkinsci/plugins/workflow/multibranch/SCMBinderTest.java
index 3e418bd..401c37a 100644
--- src/test/java/org/jenkinsci/plugins/workflow/multibranch/SCMBinderTest.java
+++ src/test/java/org/jenkinsci/plugins/workflow/multibranch/SCMBinderTest.java
@@ -48,6 +48,7 @@ import jenkins.scm.api.SCMHead;
 import jenkins.scm.api.SCMRevision;
 import jenkins.scm.api.SCMRevisionAction;
 import jenkins.scm.api.SCMSourceDescriptor;
+import jenkins.scm.impl.subversion.SubversionSCMFileSystem;
 import jenkins.scm.impl.subversion.SubversionSCMSource;
 import static org.hamcrest.Matchers.*;
 
@@ -129,6 +130,9 @@ public class SCMBinderTest {
         }
     }
 
+    static {
+        System.setProperty(SubversionSCMFileSystem.DISABLE_PROPERTY, "true");
+    }
     @Test public void exactRevisionSubversion() throws Exception {
         sampleSvnRepo.init();
         ScriptApproval sa = ScriptApproval.get();
@@ -164,7 +168,7 @@ public class SCMBinderTest {
         assertTrue(iterator.hasNext());
         ChangeLogSet.Entry entry = iterator.next();
         assertEquals("tweaked", entry.getMsg());
-        assertEquals("[/prj/trunk/Jenkinsfile, /prj/trunk/file]", new TreeSet<>(entry.getAffectedPaths()).toString());
+        assertEquals("[Jenkinsfile, file]", new TreeSet<>(entry.getAffectedPaths()).toString());
         assertFalse(iterator.hasNext());
     }
 
```

With that I think this could be mergeable, and then an analogue to https://github.com/jenkinsci/workflow-multibranch-plugin/pull/17 could be done as a follow-up.

**Comment by jglick** [comment]
BTW thanks for your patience on this! The real problem is that the Jenkins project does not consistently run https://github.com/jenkinsci/plugin-compat-tester which would allow us to detect integration test failures and genuine regressions in a timely manner, so people doing routine housekeeping wind up pulling in years’ worth of technical debt.

**Comment by slonopotamus** [comment]
I'm afraid I'm not as patient as you think.

You have #90 that passes tests, but is *a bit* hackish. You have this PR that is cleaner but exposes the fact that workflow-multibranch-plugin doesn't play together with modern subversion-plugin.

The only problem I wanted to fix - bugs that are related to updating to new parent pom and, specifically, new JTH with jenkinsci/jenkins-test-harness#166 that affected several plugins. workflow-multibranch-plugin is one of the plugins that has issues. I spent time reproducing the problem, I found the root cause, I gave a solution.

But dealing with consequences of global Jenkins code organization decisions that lead to plugins running tests against randomly-outdated versions of each other is beyound of what I am ready to handle now.

**Comment by jglick** [comment]
Ack. I can just refile this plus test fixes.

**Pull request closed without merging**